src/js/node/http.ts

```diff
@@ -1722,7 +1769,6 @@ class ClientRequest extends OutgoingMessage {
     this.#reusedSocket = false;
     this.#host = host;
     this.#protocol = protocol;
-    this.#tls = options.tls;
```

Comment:
@cirospaciari I removed this since I assumed it to be a mistake. I see that you added this option in 74d91f6b. I couldn't see any references to it in the docs, but I saw that some tests where relying on it (which I also changed to match the node API). However if we aim for Node.js compatibility, `tls` is not an option that exists, and options like `rejectUnauthorized` are expected to live directly on the `Options` object.

Reply:
Yeah, you are right this is a mistake, we should follow the docs for Node.js compatibility and pass to fetch/Bun.serve through our internal `tls` option.